zerver/management/commands/soft_activate_deactivate_users.py

```
@@ -0,0 +1,70 @@
+        self.add_realm_args(parser, True)
+        parser.add_argument('-d', '--deactivate',
+                            dest='deactivate',
+                            action='store_true',
+                            default=False,
+                            help='Used to deactivate user/users.')
+        parser.add_argument('-a', '--activate',
+                            dest='activate',
+                            action='store_true',
+                            default=False,
+                            help='Used to activate user/users.')
+        parser.add_argument('--inactive-for',
+                            type=int,
+                            default=28,
+                            help='Specify the number of days of user inactivity that user should be marked soft_deactviated')
+        parser.add_argument('users', metavar='<users>', type=str, nargs='*', default=[],
+                            help="This option can be used to specify a list of user emails to soft activate/deactivate.")
+
+    def handle(self, *args, **options):
+        # type: (*Any, **str) -> None
+        realm = self.get_realm(options)
+        user_emails = options['users']
+        activate = options['activate']
+        deactivate = options['deactivate']
+        if activate:
+            if not user_emails:
+                print('You need to specify at least one user to use the activate option.')
+                self.print_help("./manage.py", "soft_activate_deactivate_users")
+                sys.exit(1)
+            users_to_activate = list(UserProfile.objects.filter(
+                realm=realm,
+                email__in=user_emails))
+            do_soft_activate_users(users_to_activate)
+        elif deactivate:
+            if user_emails:
+                print('Soft deactivating forcefully...')
+                user_ids_to_deactivate = list(UserProfile.objects.filter(
+                    realm=realm,
+                    email__in=user_emails))
```

This should probably throw an error if the user didn't exist; this tripped me up when testing :)